### PR TITLE
highgui: change behaviour of cv::waitKey

### PR DESCRIPTION
**Pull request by alalek**
**DO NOT MERGE** Testing only.
Discussion is here: #7105

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
@alalek even though ABI compliance is an issue there, I am still considering this as an issue. The compiler will not complain with this code, but in C++ at runtime all instances of waitKey will be replaced by their inline counterpart. This is possible in C++, however Python does not support this functionality as far as I am aware.

Did you create a python test sample trying to actually use this code?

**Comment by StevenPuttemans** [comment]
Also I guess the ABI issues will always stick, because due to the inline, the code will get a changed behaviour and thus not support backwards compatibility. Therefore I still guess the double functionality is the best way to go, to avoid ABI issues also.

**Comment by StevenPuttemans** [comment]
:+1: okay seems valid then!

File: modules/highgui/include/opencv2/highgui.hpp [hunk 1]
```diff
@@ -340,7 +347,21 @@ If there are several HighGUI windows, any of them can be active.
 
 @param delay Delay in milliseconds. 0 is the special value that means "forever".
  */
+#if defined __OPENCV_BUILD
```
Comment (vpisarev) [inline modules/highgui/include/opencv2/highgui.hpp]: can we hide these details in a .cpp file?

**Pull request closed without merging**